sdk/deviceupdate/iot-device-update-rest/review/iot-device-update.api.md

```diff
@@ -0,0 +1,1709 @@
+## API Report File for "@azure-rest/iot-device-update"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { Client } from '@azure-rest/core-client';
+import { ClientOptions } from '@azure-rest/core-client';
+import { HttpResponse } from '@azure-rest/core-client';
+import { LroEngineOptions } from '@azure/core-lro';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { PathUncheckedResponse } from '@azure-rest/core-client';
+import { PollerLike } from '@azure/core-lro';
+import { PollOperationState } from '@azure/core-lro';
+import { RawHttpHeaders } from '@azure/core-rest-pipeline';
+import { RawHttpHeadersInput } from '@azure/core-rest-pipeline';
+import { RequestParameters } from '@azure-rest/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public (undocumented)
+export interface Deployment {
+    deploymentId: string;
+    groupId: string;
+    isCanceled?: boolean;
+    isRetried?: boolean;
+    startDateTime: Date | string;
```

Comment:
why is this `Date | string`? Feels weird to not just pick one, either we're deserializing it or not, since it can't be coming over the wire as a Date object

Reply:
yes, I also think it's weird to use `Date | string`, but we don't deserialize it and I don't think it can cause any issues. I will create an issue to  track this one in codegen.

Reply:
What is the actual type at runtime?

Reply:
It's string

Reply:
after chatting with @xirzec offline, it's better to fix it in codegen. So I will not change it manually.

Reply:
after chatting with @xirzec offline, it's better to fix it in codegen. So I will not change it manually.